src/lxc/conf.h

```diff
@@ -365,6 +365,9 @@ struct lxc_conf {
 
 	/* procs */
 	struct lxc_list procs;
+
+	/* path to the shared mounpoint */
+	char *shmounts_path;
```

Comment:
So I think you want two paths here:

    /* host side of shared mountpoint */
    char *shmounts_host;

    /* container side of shared mountpoint */
    char *shmounts_cont;

The use will give us `shmounts_path` via `lxc.mount.auto = inject:/shmounts/path` and we will internally figure out the `shmounts_cont` path, e.g. `/dev/.lxc-mounts`.

Might also want to make this a struct if you want:

    struct lxc_shmount {
            char *path_host;
            char *path_cont;
    };

That should also clarify the logic in `c->mount()` a little more.